packages/contracts/contracts/optimistic-ethereum/ovm/StateTransitioner.sol

```diff
@@ -229,8 +231,15 @@ contract StateTransitioner is IStateTransitioner, ContractResolver {
         stateManager.initNewTransactionExecution();
         executionManager.setStateManager(address(stateManager));
 
+        // Make sure we were given sufficient gas, accounting for EIP 150, +10000 to account for gas cost of prepping executeTransaction calldata
+        uint gasLeft = gasleft();
+        require(
+            gasLeft > ((_transactionData.gasLimit + MAX_EXECUTE_TRANSACTION_GAS_OVERHEAD + 10000) * 64 / 63),
+            "Insufficient gas supplied to ensure L1 execution will not run out of gas before OVM transaction gas limit."
```

Comment:
Would change the error message or add a comment specifically saying why the 63/64 logic is in there (maybe with a link?)